lib/Dialect/AIE/IR/AIEDialect.cpp

```diff
@@ -164,10 +164,17 @@ struct UsesAreAccessable {
     for (auto *user : users) {
       // AIE.useLock may be used in a device to set the lock's default value
       // Allow in a toplevel module for backward compatibility
-      if (llvm::isa_and_nonnull<DeviceOp, ModuleOp>(user->getParentOp()))
-        return success();
-      if (auto element = getParentTileElement(user)) {
-
+      if (llvm::isa_and_nonnull<DeviceOp, ModuleOp>(user->getParentOp())) {
+        continue;
+      }
+      // If any parent prescribes that accessibility checks be skipped,
+      // skip the check for that user.
+      if (user->getParentWithTrait<SkipAccessibilityCheckTrait>()) {
+        continue;
+      }
+      TileElement element;
+      if ((element = llvm::dyn_cast<TileElement>(user)) ||
+          (element = getParentTileElement(user))) {
```

Comment:
we could probably clean up the nesting here instead of making it more obfuscated,
```c++
for (...) {
  ...
  auto element = llvm::dyn_cast<TileElement>(user);
  if (!element)
    element = getParentTileElement(user);

  if (!element)
    return op->emitOpError("is accessed outside of a tile")

  auto tileID = element.getTileID();
  if (!targetModel.isLegalMemAffinity(tileID.col, tileID.row, thisID.col, thisID.row))
    return op->emitOpError(" is accessed from an unreachable tile in Column")
}
return success();

Reply:
Obfuscating my code is how I make myself indispensable. Just kidding I cleaned it up, thanks for the suggestion